components/PinchZoom/index.ts

```diff
@@ -0,0 +1,274 @@
+}
+
+export default class PinchZoom extends HTMLElement {
+  // The element that we'll transform.
+  // Ideally this would be shadow DOM, but we don't have the browser
+  // support yet.
+  private _positioningEl?: HTMLElement;
+  // Current transform.
+  private _x = 0;
+  private _y = 0;
+  private _scale = 1;
+
+  constructor () {
+    super();
+
+    // Watch for children changes.
+    // Note this won't fire for initial contents,
+    // so _stageElChange is also called in connectedCallback.
+    new MutationObserver(() => this._stageElChange())
+      .observe(this, { childList: true });
+
+    // Watch for pointers
+    const pointerTracker: PointerTracker = new PointerTracker(this, {
+      start: (pointer, event) => {
+        // We only want to track 2 pointers at most
+        if (pointerTracker.currentPointers.length === 2) return false;
+        event.preventDefault();
+        return true;
+      },
+      move: previousPointers => {
+        this._onPointerMove(previousPointers, pointerTracker.currentPointers);
+      }
+    });
+
+    this.addEventListener('wheel', event => this._onWheel(event));
+  }
+
+  connectedCallback () {
+    this._stageElChange();
+  }
+
+  get x () {
+    return this._x;
+  }
+
+  get y () {
+    return this._y;
+  }
+
+  get scale () {
+    return this._scale;
+  }
+
+  /**
+   * Update the stage with a given scale/x/y.
+   */
+  setTransform (scale: number = this._scale, x: number = this._x, y: number = this._y, {
+    allowChangeEvent = false
+  }: {
+    /**
+     * Fire a 'change' event if values are different to current values
+     */
+    allowChangeEvent?: boolean
+  } = {}) {
+    // Bail if there's no change
+    if (
+      scale === this._scale &&
+      x === this._x &&
+      y === this._y
+    ) return;
+
+    this._x = x;
+    this._y = y;
+    this._scale = scale;
+
+    this.style.setProperty('--x', this._x + 'px');
+    this.style.setProperty('--y', this._y + 'px');
+    this.style.setProperty('--scale', this._scale + '');
+
+    if (allowChangeEvent) {
+      const event = new Event('change', { bubbles: true });
+      this.dispatchEvent(event);
+    }
+  }
+
+  /**
+   * Called when the direct children of this element change.
+   * Until we have have shadow dom support across the board, we
+   * require a single element to be the child of <pinch-zoom>, and
+   * that's the element we pan/scale.
+   */
+  private _stageElChange () {
+    this._positioningEl = undefined;
+
+    if (this.children.length === 0) {
+      console.warn('There should be at least one child in <pinch-zoom>.');
+      return;
+    }
+
+    const el = this.children[0];
+
+    if (!(el instanceof HTMLElement)) {
+      console.warn('The first child of <pinch-zoom> must be an HTMLElement.');
+      return;
+    }
+
+    this._positioningEl = el;
+
+    if (this.children.length > 1) {
+      console.warn('<pinch-zoom> must not have more than one child.');
+    }
+
+    this.setTransform();
```

Comment:
But... calling `setTransform` with the default values means it will just bail, did you mean to do something else here?

Reply:
I might still need this, but some stuff has gone awry in a refactor.

I need to:

* Ensure bounds are respected when calling `setTransform`.
* `setTransform` should be callable before adding `<pinch-zoom>` to the document, but bounds will need to be rechecked once it's added to the document.